### PR TITLE
Add blog list to our home page

### PR DESCRIPTION
**Pull request by choldgraf**
This adds a lightweight list of recent blog posts to our home page. It uses the [feedparser](https://feedparser.readthedocs.io/en/latest/common-atom-elements.html) package to parse the XML feed of the Jupyter Blog that has the `jupyterhub` topic. This means **if anybody tags a blog post with `jupyterhub`, it will show up here**.

- It downloads the latest pages at build time, and then inserts them into the home page. We can iterate on that over time but just wanted to keep it simple for now.
- It also adds a CRON job to build these docs every day to keep this list up to date.
- It adds `jupyterhub_sphinx_theme` as an **extension** because this this is needed to properly set configuration defaults.

### CODE REVIEW

**Comment by choldgraf** [comment]
cc @sgibson91 who discussed this in one of the team compass issues I believe!

**Review by minrk — APPROVED** [review]

**Review by sgibson91 — APPROVED** [review]
Thanks Chris, this is super cool! Glad to have an example of this

File: docs/_data/blog-posts.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- [Introducing JupyterHub’s Outreachy interns! — December 2022 Cohort](https://blog.jupyter.org/introducing-jupyterhubs-outreachy-interns-december-2022-cohort-23aaf4613556?source=rss----95916e268740--jupyterhub) by **Sarah Gibson**
```
Comment (sgibson91) [inline docs/_data/blog-posts.txt]: Should we gitignore this file if it's generated at build time?

Comment (choldgraf) [reply docs/_data/blog-posts.txt]: Oops yep - I wanted to fit keep the folder but forgot to gitignore everything inside.

I'll plan to fix that and then merge 👍

**Comment by choldgraf** [comment]
OK we're now ignoring the blog post auto-generated text! I'll merge this one since we have two 👍 s

**Pull request merged**